### DISCUSSION TITLE
Pass environment variables in docker-compose

### DISCUSSION THREAD

**Discussion by Koliham** [quasarframework/quasar#12955] [General - Components / Directives / etc]
I want to run 2 instances of my docker image of the quasar app, but with different environment variables I would like to pass in the docker-compose file.
The environment variables are within the docker container itself (echo $MYVAR), but are not in the SPA itself.
My question is pretty much the same as here:
https://forum.quasar-framework.org/topic/6798/quasar-can-t-access-environment-variables-in-docker-container?_=1648418331246&lang=de

**Comment by straurob** [comment]
I asked the same question on Stack Overflow and this answer was helpful for me:

https://stackoverflow.com/questions/72147171/pass-docker-environment-variable-into-vue-quasar-application-at-runtime/72153363#72153363

**Reply by cdecinkoKnight** [reply]
These feels very, very dirty, putting the values of the environment variables right out in the open inside the index.html head. The point of secrets is to keep them secret. Is there not a better way?
